examples/vector_return/src/lib.rs

```diff
@@ -0,0 +1,30 @@
+extern crate libc;
+
+use libc::size_t;
+use std::mem;
+
+#[no_mangle]
+pub extern fn counter_generate(start: size_t, size: size_t, vec: *mut *mut size_t) -> size_t {
```

Comment:
I'd rather a type other than `size_t` for the element since it's easy to lose it compared to the `size` parameter.